### PR TITLE
Dev/search filter

### PR DESCRIPTION
**Pull request by noahboegli**
The filter is now available on GET routes with no ID.

List of operations supported

- where with nested filed (caution: entity name, not DTO name): `/inventory?where[owner][firstName][$eq]=Paul`
   - The supported comparison (always required are): `eq`, `neq`, `lt`, `gt`, `gte`, `lte`
   - Where can be chained in an _AND_ manner (_OR_ could be supported, if required): `/inventory?where[owner][firstName][$eq]=Paul&where[id][$lte]=4`
- limit/offset: `/inventory?limit=5&offset=5`
- A sort option could be implemented

### CODE REVIEW

**Comment by noahboegli** [comment]
@HugoMendes98  I hope I managed to hold to your query filter standards, or at least come close to them.

File: apps/backend/src/main/java/ch/heig/pdg/backend/BackendApplication.java [hunk 1]
```diff
@@ -29,6 +32,15 @@ public JsonNullableModule jsonNullableModule() {
         return new JsonNullableModule();
     }
 
+    @Bean
```
Comment (nigelmann) [inline apps/backend/src/main/java/ch/heig/pdg/backend/BackendApplication.java:35]: You like beans ?
https://youtu.be/E-GD1_0MUGw?t=119

File: apps/backend/src/main/java/ch/heig/pdg/backend/BackendApplication.java [hunk 2]
```diff
@@ -29,6 +32,15 @@ public JsonNullableModule jsonNullableModule() {
...
+    public WebServerFactoryCustomizer<TomcatServletWebServerFactory> tomcatCustomizer() {
+        return (tomcat) ->
+                tomcat.addConnectorCustomizers((connector) ->
+                        ((AbstractHttp11Protocol<?>)connector.getProtocolHandler()).setRelaxedQueryChars("[]")
+        );
+    }
+
+
```
Comment (nigelmann) [inline apps/backend/src/main/java/ch/heig/pdg/backend/BackendApplication.java:43]: And you were the one criticizing my line breaks

**Review by nigelmann — COMMENTED** [review]
I think we should merge checkstyle into this branch and then merge it into develop, any thoughts ?

**Comment by HugoMendes98** [comment]
> @HugoMendes98 I hope I managed to hold to your query filter standards, or at least come close to them.

@noahboegli Close enough :).

Some feedback:
- Do the comparison operators support dates? Needed if we want to show only some items after a point in time.
  - And what format do they expect? The standard ISO (`2000-01-01T12:00:00.000Z`)?
- Given the current developed features, a `OR` filter does not seem needed at the moment.
- What happens with a `limit=0`? Does it fail or return nothing (~count only)?
- A sorting option is necessary; the application is first and foremost a list of items. If they can't be ordered, users have less reason to use it.
  - An `order` object property would be ideal (ex: `order[0][id]=asc&order[1][name]=desc`), however, given the time limit, a simplier solution can be implemented (ex: `order=id,-name`, the `-` meaning _desc_).  
  - I let you decide what's the easier.

**Pull request merged**